src/frame_publisher.cpp

```diff
@@ -52,58 +53,59 @@ bool FramePublisher::initialize()
 {
   priv_nh_ = ros::NodeHandle("~");
 
-  priv_nh_.param<double>("update_rate", update_rate_, 0.01); // 100Hz
+  priv_nh_.param<double>("update_rate", update_rate_, 0.01);  // 100Hz
 
-  priv_nh_.param<std::string>("base_frame", base_frame_, "base_link");
-  priv_nh_.param<std::string>("rotation_frame", rotation_frame_, "torso_center_link");
-  priv_nh_.param<std::string>("target_frame", target_frame_, "torso_rotated_base_link");
+  priv_nh_.param<std::string>("from_frame", from_frame_, "base_link");
+  priv_nh_.param<std::string>("to_frame", to_frame_, "torso_center_link");
+  priv_nh_.param<std::string>("frame_name", frame_name_, "torso_rotated_base_link");
+
+  priv_nh_.param<bool>("use_to_frame_translation", use_to_frame_translation_, true);
 
   priv_nh_.param<bool>("rot_z", rot_z_, false);
   priv_nh_.param<bool>("rot_x", rot_x_, false);
   priv_nh_.param<bool>("rot_y", rot_y_, false);
 
   frame_broadcast_timer_ = nh_.createTimer(ros::Duration(update_rate_), &FramePublisher::frameBroadcastCallback, this);
 
-  ros::Duration(1.0).sleep(); //give tf_listener some time
+  ros::Duration(1.0).sleep();  // give tf_listener some time
 
   return true;
 }
 
-/// Broadcast a frame aligned with the base frame but rotated around specified axes as rotation_frame 
+/// Broadcast a frame aligned with the base frame but rotated around specified axes as rotation_frame
 void FramePublisher::frameBroadcastCallback(const ros::TimerEvent& event)
 {
   ros::Time time = event.current_real;
   geometry_msgs::TransformStamped transform_msg;
-  try{
-    transform_msg = tf_buffer_.lookupTransform(base_frame_, rotation_frame_, time, ros::Duration(0.1));
+  try
+  {
+    transform_msg = tf_buffer_.lookupTransform(from_frame_, to_frame_, time, ros::Duration(0.1));
```

Comment:
we might need to use `ros::Time(0)` here....
@lindemeier please let me know in case you face problems with `event.current_real`